server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3469,8 +3469,31 @@ private void doSnapshotShard(SnapshotShardContext context) {
     private static void ensureNotAborted(ShardId shardId, SnapshotId snapshotId, IndexShardSnapshotStatus snapshotStatus, String fileName) {
         try {
             snapshotStatus.ensureNotAborted();
+
+            if (snapshotStatus.getStage() != IndexShardSnapshotStatus.Stage.INIT
+                && snapshotStatus.getStage() != IndexShardSnapshotStatus.Stage.STARTED) {
+                // A normally running shard snapshot should be in stage INIT or STARTED. And we know it's not in PAUSING or ABORTED because
+                // the ensureNotAborted() call above did not throw. The remaining options don't make sense, if they ever happen.
+                logger.info(
```

Comment:
If this is unexpected, `info` seems wrong, I think this should perhaps be `error`? Maybe with an `assert false` too?